bccsp/pkcs11/pkcs11.go

```diff
@@ -84,25 +84,44 @@ func loadLib(lib, pin, label string) (*pkcs11.Ctx, uint, *pkcs11.SessionHandle,
 func (csp *impl) getSession() (session pkcs11.SessionHandle) {
 	select {
 	case session = <-csp.sessions:
-		logger.Debugf("Reusing existing pkcs11 session %+v on slot %d\n", session, csp.slot)
+		_, err := csp.ctx.GetSessionInfo(session)
+		if err != nil {
+			logger.Warningf("Get session info failed [%s], getting new session", err)
```

Comment:
You missed a `\n` character at the end of this. But I also don't think the message conveys what the check is actually doing, maybe `Existing session invalidated, creating a new session\n`, I'm not sure adding the error would provide value to the user, as we've already once established this session, so it's not something the user did that caused the session to fail.